app/controllers/api/v1/mixins/service_offering_mixin.rb

```diff
@@ -4,13 +4,22 @@ module Mixins
       module ServiceOfferingMixin
         private
 
-        def service_offering_check(order)
-          service_offering_service = Catalog::ServiceOffering.new(order).process
+        def service_offering_check
+          service_offering_service = Catalog::ServiceOffering.new(@order).process
           if service_offering_service.archived
-            raise Catalog::ServiceOfferingArchived, "Service offering for order #{order.id} has been archived and can no longer be ordered"
+            @order.order_items.each do |order_item|
+              order_item.update!(:completed_at => DateTime.now, :state => "Failed")
+              order_item.update_message("error", archived_error_message)
+            end
+
+            @order.update!(:state => "Failed")
+
+            raise Catalog::ServiceOfferingArchived, archived_error_message
           end
+        end
 
-          true
+        def archived_error_message
+          "Order Failed: Service offering for order #{@order.id} has been archived and can no longer be ordered"
```

Comment:
@eclarizio Does it make sense to include the name of the service offering here along with the order id.

Reply:
We could, but we currently don't have access to that object, we only have access to the `archived` property. If you want me to add it, I can, it's a simple change.